pkg/operator/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_secret_reconcile.go

```diff
@@ -0,0 +1,89 @@
+package clustermanagercontroller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/kubernetes"
+
+	operatorapiv1 "open-cluster-management.io/api/operator/v1"
+
+	"open-cluster-management.io/ocm/manifests"
+	"open-cluster-management.io/ocm/pkg/operator/helpers"
+)
+
+const (
+	// workDriverConfig is the secret that contains the work driver configurarion
+	workDriverConfig = "work-driver-config"
+)
+
+var (
+	// secretNames is the set of secrets to be synced from source namespace to the target namespace
+	secretNames = sets.New[string]()
+)
+
+type secretReconcile struct {
+	operatorKubeClient kubernetes.Interface
+	hubKubeClient      kubernetes.Interface
+	operatorNamespace  string
+	cache              resourceapply.ResourceCache
+	recorder           events.Recorder
+}
+
+func (c *secretReconcile) reconcile(ctx context.Context, cm *operatorapiv1.ClusterManager,
+	config manifests.HubConfig) (*operatorapiv1.ClusterManager, reconcileState, error) {
+	if config.CloudEventsDriverEnabled && config.WorkDriver != string(operatorapiv1.WorkDriverTypeKube) {
+		secretNames = secretNames.Insert(workDriverConfig)
+	} else {
+		secretNames = secretNames.Delete(workDriverConfig)
+	}
+
+	for _, secretName := range secretNames.UnsortedList() {
+		// check the source secret explicitly as the 'helpers.SyncSecret' doesn't return an error
+		// when the source secret is not found.
+		if _, err := c.operatorKubeClient.CoreV1().Secrets(c.operatorNamespace).Get(ctx,
```

Comment:
let's remove this additional get for now. It could happen this should return success if the secret does not exist

Reply:
removed this.